src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -278,6 +281,54 @@ public static bool All(this IEnumerable<bool> source)
             return true;
         }
 
+        public static int IndexOf<T>(this IEnumerable<T> sequence, T value)
+        {
+            return sequence switch
+            {
+                IList<T> list => list.IndexOf(value),
+                IReadOnlyList<T> readOnlyList => IndexOf(readOnlyList, value, EqualityComparer<T>.Default),
+                _ => EnumeratingIndexOf(sequence, value, EqualityComparer<T>.Default)
+            };
+        }
+
+        public static int IndexOf<T>(this IEnumerable<T> sequence, T value, IEqualityComparer<T> comparer)
+        {
+            return sequence switch
+            {
+                IReadOnlyList<T> readOnlyList => IndexOf(readOnlyList, value, comparer),
```

Comment:
No IList variant here? I know IList<T>.IndexOf exists but if somebody has an enumerable that is under the covers an IList<T> is it still worth delegating? Just like you did for the prior method?

Reply:
It doesn't have IndexOf that takes comparer.